EventTicketingSystem.DataAccess/Services/CityRepository.cs

```diff
@@ -0,0 +1,52 @@
+﻿using EventTicketingSystem.DataAccess.Interfaces;
+using EventTicketingSystem.DataAccess.Models.Context;
+using EventTicketingSystem.DataAccess.Models.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace EventTicketingSystem.DataAccess.Services
+{
+
+    internal class CityRepository : ICityRepository
+    {
+        private readonly DatabaseContext _context;
+
+        public CityRepository(DatabaseContext context)
+        {
+            _context = context;
+        }
+        public async Task<ICollection<City>> GetCities()
+        {
+            return await _context.Cities.ToListAsync();
+        }
+
+        public async Task<City> Find(int id)
+        {
+            return await _context.Cities.FindAsync(id);
+        }
+
+        public async Task Add(City city)
+        {
+            await _context.AddAsync(city);
```

Comment:
Looks like CUD operations can be moved out to the base class with dataset gathering from the context(generic feature)

Reply:
Done